src/compiler/function.ts

```diff
@@ -78,10 +78,17 @@ export function isFunctionExpressionMethod(node: ts.FunctionExpression) {
 }
 
 export function isMethodDeclaration(node: ts.Node<ts.ts.Node>): node is ts.MethodDeclaration | ts.FunctionExpression {
-	return (
+	if (
+		ts.TypeGuards.isMethodSignature(node) ||
```

Comment:
This was the culprit. Added this because declared methods weren't being seen as methods. I'm guessing that this is fine?